app/javascript/application.js

```diff
@@ -33,3 +34,5 @@ const ReactRailsUJS = require('react_ujs');
 ReactRailsUJS.useContext(componentRequireContext);
 
 Turbo.session.drive = false;
+
+mermaid.initialize({ startOnLoad: true, theme: 'dark' });
```

Comment:
I moved it to a Stimulus controller, but the tests are still being weird :thinking:

Reply:
I've been trying some stuff out on a test branch, theres no good way of debugging CI locally that I'm aware of 😭 
I've narrowed it down to the imports but that doesn't tell us a lot.

It compiles fine locally and on the review app, so it has got to be some issue with our CI config. Will try digging into more over the weekend

Reply:
Thanks for looking into it. From some light googling, I did find several ideas from searching "can't find application.css in /app/public/packs-test/manifest.json" and looking for cases that best match what's happening here. I've been able to find some issues that seem similar enough (Rails 6, occurring in CI/CD pipeline, empty `manifest.json`). Some examples that may help:
- https://github.com/rails/webpacker/issues/2674
- https://forum.gitlab.com/t/webpacker-cant-find-application-in-app-public-packs-test-manifest-json/31311
- https://jumpstartrails.com/discussions/webpacker-can-t-find-application-js-in-app-public-packs-manifest-json
- https://anthonybroadcrawford.com/2020/03/02/webpacker-heroku-ci.html

Reply:
Thanks for those links Josh, everything suggests we need an assets compilation step somewhere in the circle config. No luck getting that working yet, but I'll keep at it.

But whats _really_ got me scratching my head with this... we have react components all working fine with system tests on CI without any explicit asset compilation happening and they have a much more complex config than this 😅

Reply:
Discussed this with crespire some and he managed to find this: https://github.com/shakacode/shakapacker/issues/96 -- which seems like a pretty promising discussion around this problem. Hope it helps!

Reply:
Thank you for that issue, it was very useful with debugging this.

The issue is our webpack config doesn't know how to handle the mermaid import in test mode. 
<img width="1120" alt="Screenshot 2023-04-01 at 03 06 38" src="https://user-images.githubusercontent.com/7963776/229261621-ab1d3f07-84c0-4e0d-a0fb-5820950ff544.png">

Mermaid switched to [ESM only in version 10](https://github.com/mermaid-js/mermaid/blob/master/CHANGELOG.md#1000) which changed their exports script for the package.

Our Webpack config probably needs updated to work with ESM only packages to get this to work. But my Webpack knowledge is very limited 😢  and I couldn't find anything very useful when searching. 

Another option, would be to downgrade to mermaid 9.4.3, which works fine with CI. We could then put together an issue for someone with more Webpack experience to update the test config and upgrade to v10.

Reply:
I put in a commit that downgrades to 9.4.3. Had to put in a small change in the `LessonPreview` react component because of some API changes, but other than that it seems okay. And it passes the CI checks!

And yeah, I'm *definitely* no Webpack expert lol. If there's any hesitancy on merging this knowing that it could be annoying to update later, I definitely get that. Would be a neat tool to have though I think!

Reply:
I did have one question, do you have a way of creating the failures happening in CI locally? I have a couple of ideas I might want to try, but it'd be nice if I didn't have to go through CI to see if things work or not.

Reply:
No worries, I'm happy for it to go in with the downgraded version. It _probably_ only needs a simple Webpack config change to get the latest version working.

I wasn't able to reproduce locally unfortunately. Running the webpack compile command manually (`RAILS_ENV=test rails webpacker:compile`) succeeds locally but fails in CI.

Reply:
I think I'm good to merge. I didn't find anything that fixes it, but I did find that with `NODE_ENV=test RAILS_ENV=test rails webpacker:compile`, the errors will be reproduced locally while on mermaid v10. This should at least help whoever wants to pick up the issue to get it working with v10.

Also I'm definitely willing to put together an Issue on bumping it to v10 if you want.

Reply:
Nice one! thanks @JoshDevHub, thats going to help a lot.

Thanks for offering to put the issue together mate, I'll definitely take you up on that aha.